js/interface.js

```diff
@@ -30,13 +31,25 @@ $('form').submit(function (event) {
 // 3. Fired when the provider has finished
 linkActionProvider.then(function (result) {
   data.action = result.data;
+  save(true);
+});
+
+function save(notifyComplete) {
   data.label = $('#primaryButtonLabel').val();
 
   Fliplet.Widget.save(data).then(function () {
-    Fliplet.Widget.complete();
-    window.location.reload();
+    if (notifyComplete) {
+      Fliplet.Widget.complete();
+      window.location.reload();
+    } else {
+      Fliplet.Studio.emit('reload-widget-instance', widgetId);
+    }
   });
-});
+}
+
+$('#primaryButtonLabel').keyup($.debounce(function() {
```

Comment:
I would include other events. Typically I'd also include: `mousedown`, `paste`, `input` and `change`